### PR TITLE
Mydevel eta comments

### PR DESCRIPTION
**Pull request by gkaratha**

### CODE REVIEW

File: L1Trigger/L1TMuonBarrel/src/L1MuBMTQualPatternLut.cc [hunk 1]
```diff
@@ -215,8 +218,24 @@ const vector<short>& L1MuBMTQualPatternLut::getQualifiedPatterns(int sp, int adr
 
   LUT::const_iterator it = m_lut.find(make_pair(sp,adr));
   if ( it == m_lut.end() ) {
-    cerr << "Error: L1MuDTQualPatternLut: no pattern list found for address " << adr << endl;
+    edm::LogError ("Not found") << "Error: L1MuDTQualPatternLut: no pattern list found for address " << adr << endl;
   }
```
Comment (thomreis) [inline L1Trigger/L1TMuonBarrel/src/L1MuBMTQualPatternLut.cc]: Can you be a bit more specific? Not found is a bit generic. Normally, the module name goes there.

File: L1Trigger/L1TMuonBarrel/src/L1MuBMTQualPatternLut.cc [hunk 2]
```diff
@@ -215,8 +218,24 @@ const vector<short>& L1MuBMTQualPatternLut::getQualifiedPatterns(int sp, int adr
...
+     string str=file.readString();
+     if (str.find("#")==0) skip+=1;
+     // cout<<"here "<<str<<" found "<<str.find("#")<<endl;
+        if ( !file.good() ) { file.close(); break; }
+    
+  }
+  // cout<<"coarse lines to skip3 "<<skip<<endl;
+  file.close();
+  return skip;
+}
```
Comment (thomreis) [inline L1Trigger/L1TMuonBarrel/src/L1MuBMTQualPatternLut.cc]: @gkaratha can you reuse the function already defined in L1MuBMTEtaPatternLut? They seem to be the same.

File: L1Trigger/L1TMuonBarrel/src/L1MuBMTEtaPatternLut.cc [hunk 3]
```diff
@@ -101,11 +102,14 @@ cout<<"skata"<<endl;
   //                                       << file.getName() << endl; 
 
   // ignore comment lines 
-  file.ignoreLines(16);
  
+  int skip2=getIgnoredLines(file);
+  // cout<<" lines to skip2 "<<skip2<<endl;
+    file.ignoreLines(skip2);
+
```
Comment (thomreis) [inline L1Trigger/L1TMuonBarrel/src/L1MuBMTEtaPatternLut.cc:109]: What do you do if there is a comment not at the beginning of the file? Can that happen?

File: L1Trigger/L1TMuonBarrel/src/L1MuBMTEtaPatternLut.cc [hunk 4]
```diff
@@ -89,7 +90,7 @@ int L1MuBMTEtaPatternLut::load() {
   // get directory name
   string defaultPath = "L1Trigger/";//L1TriggerConfig/DTTrackFinder/parameters/";
   string eau_dir = "L1TMuon/data/bmtf_luts/LUTs_Ass/";//L1TriggerData/DTTrackFinder/Eau/
-cout<<"skata"<<endl;
+
```
Comment (thomreis) [inline L1Trigger/L1TMuonBarrel/src/L1MuBMTEtaPatternLut.cc]: I guess the commented out paths can be removed.

File: L1Trigger/L1TMuonBarrel/src/L1MuBMAssignmentUnit.cc [hunk 5]
```diff
@@ -175,7 +175,7 @@ void L1MuBMAssignmentUnit::PhiAU(const edm::EventSetup& c) {
   const double k = 57.2958/0.625/static_cast<float>(phi_precision);
   double phi_f = static_cast<double>(phi2);
    int bit_div_phi=static_cast<int>(phi2)%4;
```
Comment (thomreis) [inline L1Trigger/L1TMuonBarrel/src/L1MuBMAssignmentUnit.cc]: If you are fixing indents check also this one.

File: L1Trigger/L1TMuonBarrel/src/L1MuBMTrackFinder.cc [hunk 6]
```diff
@@ -211,9 +211,9 @@ void L1MuBMTrackFinder::run(const edm::Event& e, const edm::EventSetup& c) {
 
             l1t::RegionalMuonCand rmc;
 
-            if(cand->hwEta()>-117 || cand->hwEta()<117 )
+            if(cand->hwEta()>-107 || cand->hwEta()<107 )//some upper and lower values for eta
                 rmc.setHwEta(cand->hwEta());
```
Comment (thomreis) [inline L1Trigger/L1TMuonBarrel/src/L1MuBMTrackFinder.cc:215]: Why do you change this to 107 now? What was wrong with 117?

**Review by thomreis — COMMENTED** [review]
Why do you need to restore the LUTs in  `L1TriggerConfig/DTTrackFinder/parameters/L1TriggerData/DTTrackFinder/Eau/` again? Was this not done in #558 already?

File: L1Trigger/L1TMuonBarrel/src/L1MuBMTQualPatternLut.cc (hunk 1 above)
Comment (gkaratha) [reply L1Trigger/L1TMuonBarrel/src/L1MuBMTQualPatternLut.cc]: changed in new PR

File: L1Trigger/L1TMuonBarrel/src/L1MuBMTQualPatternLut.cc (hunk 2 above)
Comment (gkaratha) [reply L1Trigger/L1TMuonBarrel/src/L1MuBMTQualPatternLut.cc]: changed in the new PR

File: L1Trigger/L1TMuonBarrel/src/L1MuBMTEtaPatternLut.cc (hunk 4 above)
Comment (gkaratha) [reply L1Trigger/L1TMuonBarrel/src/L1MuBMTEtaPatternLut.cc]: changed in new PR

File: L1Trigger/L1TMuonBarrel/src/L1MuBMTEtaPatternLut.cc (hunk 3 above)
Comment (gkaratha) [reply L1Trigger/L1TMuonBarrel/src/L1MuBMTEtaPatternLut.cc:109]: We do not. To read the LUT we use readString and readInteger. A comment not in the beginning could cause crash if it is on coarse and only coarse values if it is in fine.

File: L1Trigger/L1TMuonBarrel/src/L1MuBMAssignmentUnit.cc (hunk 5 above)
Comment (gkaratha) [reply L1Trigger/L1TMuonBarrel/src/L1MuBMAssignmentUnit.cc]: fixed in new PR

File: L1Trigger/L1TMuonBarrel/src/L1MuBMTrackFinder.cc (hunk 6 above)
Comment (gkaratha) [reply L1Trigger/L1TMuonBarrel/src/L1MuBMTrackFinder.cc:215]: LUTs of eta give lower values than 107. 107 is better suited. (-106,106) fine and (-104,104) coarse

File: L1Trigger/L1TMuonBarrel/src/L1MuBMAssignmentUnit.cc [hunk 7]
```diff
@@ -196,8 +196,8 @@ void L1MuBMAssignmentUnit::PhiAU(const edm::EventSetup& c) {
   phi_8 += sectordiff*48;
 
   int phi = phi_8 + 24;
-  if (phi >  60) phi =  60;
-  if (phi < -8) phi = -8;
+  if (phi >  69) phi =  69;//upper bound on phi assignment - phi is calculated in emulator all muons are<=60 but theoretically we can go up to 69
```
Comment (rekovic) [inline L1Trigger/L1TMuonBarrel/src/L1MuBMAssignmentUnit.cc:199]: @gkaratha 
I am sorry, but I don't understand this.  Please rephrase. 

Why was this value "60" before?  Was this a bug?
Where does the the number "69" come from ?

Comment (rekovic) [reply L1Trigger/L1TMuonBarrel/src/L1MuBMAssignmentUnit.cc:199]: @gkaratha 
Again, I don't understand your comment.
Was 117 a bug?   Where did 117 come from?
Where does 107 come from?  I understand that 107 is grater than 106, but what is 106? Where
do these numbers come from ?

**Comment by rekovic** [comment]
@gkaratha To proceed with my PR #19102, please first explain better where these numbers 
"60" --> "69" 
"117" --> "107" 
came from.  Thanks.

**Comment by thomreis** [comment]
@rekovic this can be closed now I guess?

**Comment by rekovic** [comment]
comments addressed directly to CMSSW PR https://github.com/cms-sw/cmssw/pull/19102
and subsequently applied to the l1t-integration branch.  So this can be closed.

**Pull request closed without merging**